From cdd786824f970006e6f051b8c1a8bab82fd7bccf Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Sat, 2 Dec 2023 22:59:19 +0100
Subject: [PATCH 0807/1141] media: dcmipp: byteproc: perform all init before
 subdev_register

The subdev must be fully initialized when the subdev registration is
done since it becomes usable as soon as registration is done.

Change-Id: Iddb0e3ce155c662c57147f2de96d7ff0db0e54c6
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/344616
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../st/stm32/stm32-dcmipp/dcmipp-byteproc.c      | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c
index 4db4b8934137..feae08de9881 100644
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c
@@ -783,10 +783,18 @@ static int dcmipp_byteproc_comp_bind(struct device *comp, struct device *master,
 		return -ENOMEM;
 
 	byteproc->regs = bind_data->regs;
+	byteproc->dev = comp;
 
 	/* Initialize the lock */
 	mutex_init(&byteproc->lock);
 
+	/* Initialize the frame format */
+	byteproc->sink_fmt = fmt_default;
+	byteproc->crop = r;
+	byteproc->compose = r;
+	byteproc->src_interval = interval;
+	byteproc->sink_interval = interval;
+
 	/* Initialize ved and sd */
 	ret = dcmipp_ent_sd_register(&byteproc->ved, &byteproc->sd,
 				     bind_data->v4l2_dev,
@@ -805,14 +813,6 @@ static int dcmipp_byteproc_comp_bind(struct device *comp, struct device *master,
 	}
 
 	dev_set_drvdata(comp, &byteproc->ved);
-	byteproc->dev = comp;
-
-	/* Initialize the frame format */
-	byteproc->sink_fmt = fmt_default;
-	byteproc->crop = r;
-	byteproc->compose = r;
-	byteproc->src_interval = interval;
-	byteproc->sink_interval = interval;
 
 	return 0;
 }
-- 
2.39.2

